src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -1994,6 +1998,16 @@ def get_multisite_sync_status(self):
             raise DashboardException(error, http_status_code=500, component='rgw')
         return {}
 
+    def get_daemon_realm_name(self, daemon_name: str) -> str:
+        for _, server in CephService.get_service_map('rgw').items():
+            for service in server['services']:
+                metadata = service['metadata']
+                if metadata['id'] == daemon_name:
+                    return metadata['realm_name']
```

Comment:
Consider using `next` as search stops as soon as item found and easier to read as well.
Similar to what you are doing here.